### PR TITLE
GP-9257 Fix parameter mismatch in report notification

### PR DESCRIPTION
**Pull request by pfigel**
This fixes an issue where `Notification\Report` expects an `AdyenNotification` entity array, but receives the original Adyen payload instead. This causes the `event_date` not to be set (as the relevant payload value is called `eventDate` instead), causing empty `cancel_date` values.

### CODE REVIEW

File: tests/phpunit/Civi/Adyen/Report/Processor/SettlementDetailTest.php [hunk 1]
```diff
@@ -135,7 +135,7 @@ public function testSettlement() {
       $this->createReportLine(1, []),
       $this->report
     );
-    $reportProcessor->process();
+    $this->assertTrue($reportProcessor->process());
```
Comment (pfigel) [inline tests/phpunit/Civi/Adyen/Report/Processor/SettlementDetailTest.php:138]: Unrelated change - just figured we might as well check that processing worked.

**Review by mflandorfer — APPROVED** [review]
